utils/utils.go

```diff
@@ -37,16 +37,27 @@ func MarshalToFile(data any, filePath string) error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(filePath, marshaled, 0o644)
+	return os.WriteFile(filePath, marshaled, common.FilePerm644)
+}
+
+// MarshalToFileReadOnlyByRoot file only readable by root,
+// use this during process of running an Image Based Upgrade for saving cluster information whenver possible
+func MarshalToFileReadOnlyByRoot(data any, filePath string) error {
+	marshaled, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(filePath, marshaled, common.FilePerm600)
 }
 
 // MarshalToYamlFile marshals any object to YAML and writes it to the given file path
+// file only readable by root
```

Comment:
same as above function ?

Reply:
Yeah i thought we would needed to split the write func depending if it's "non-ibu" component or not. But change is applicable globally. Removed it now